src/io_header.c

```diff
@@ -0,0 +1,480 @@
+  mq1_field->exposure_time_s[index] = mq1_h.exposure_time_s;
+  mq1_field->counter[index]         = mq1_h.counter;
+  mq1_field->colour_mode[index]     = mq1_h.colour_mode;
+  mq1_field->gain_mode[index]       = mq1_h.gain_mode;
+  /*threshold is float[8]*/
+  for (unsigned int i = 0; i < MQ1_FLOAT_LEN_THRESHOLD; ++i) {
+    mq1_field->threshold[index * MQ1_FLOAT_LEN_THRESHOLD + i] =
+      mq1_h.threshold[i];
+  }
+  /*header_extension_id is char[5]*/
+  snprintf(mq1_field->header_extension_id +
+             index * MQ1_CHAR_LEN_HEADER_EXTENSION_ID,
+           MQ1_CHAR_LEN_HEADER_EXTENSION_ID, "%s", mq1_h.header_extension_id);
+  /*extended_timestamp is char[31]*/
+  snprintf(mq1_field->extended_timestamp +
+             index * MQ1_CHAR_LEN_EXTENDED_TIMESTAMP,
+           MQ1_CHAR_LEN_EXTENDED_TIMESTAMP, "%s", mq1_h.extended_timestamp);
+  mq1_field->exposure_time_ns[index] = mq1_h.exposure_time_ns;
+  mq1_field->bit_depth[index]        = mq1_h.bit_depth;
+}
+
+/* Mainly this is just for easy access to mq1_fields and dac
+ */
+
+info *mq1_fields_info(MQ1_fields *fields_struct, size_t num_count)
+{
+  if (!fields_struct || !num_count)
+    return NULL;
+
+  if (num_count != 19) {
+    fprintf(stderr, "Wrong number of fields in mq1_fields_info\n");
+    return NULL;
+  }
+
+  info *fields = malloc(num_count * sizeof(info));
+  if (!fields)
+    return NULL;
+
+  fields[0]  = (info) {"max_length", &fields_struct->max_length};
+  fields[1]  = (info) {"sequence_number", fields_struct->sequence_number};
+  fields[2]  = (info) {"header_bytes", fields_struct->header_bytes};
+  fields[3]  = (info) {"num_chips", fields_struct->num_chips};
+  fields[4]  = (info) {"det_x", fields_struct->det_x};
+  fields[5]  = (info) {"det_y", fields_struct->det_y};
+  fields[6]  = (info) {"pixel_depth", fields_struct->pixel_depth};
+  fields[7]  = (info) {"sensor_layout", fields_struct->sensor_layout};
+  fields[8]  = (info) {"chip_select", fields_struct->chip_select};
+  fields[9]  = (info) {"timestamp", fields_struct->timestamp};
+  fields[10] = (info) {"exposure_time_s", fields_struct->exposure_time_s};
+  fields[11] = (info) {"counter", fields_struct->counter};
+  fields[12] = (info) {"colour_mode", fields_struct->colour_mode};
+  fields[13] = (info) {"gain_mode", fields_struct->gain_mode};
+  fields[14] = (info) {"threshold", fields_struct->threshold};
+  fields[15] =
+    (info) {"header_extension_id", fields_struct->header_extension_id};
+  fields[16] = (info) {"extended_timestamp", fields_struct->extended_timestamp};
+  fields[17] = (info) {"exposure_time_ns", fields_struct->exposure_time_ns};
+  fields[18] = (info) {"bit_depth", fields_struct->bit_depth};
+
+  return fields;
+}
+
+info *dac_info(dac_rx *dac, size_t num_count)
+{
+  if (!dac || !num_count)
+    return NULL;
+
+  if (num_count != 28) {
+    fprintf(stderr, "Wrong number of fields in dac_info\n");
+    return NULL;
+  }
+
+  info *dinfo = malloc(NUM_FIELDS * sizeof(info));
```

Comment:
The macros `NUM_FIELDS` is not defined? And `mq1_fields_info` doesn't use a macros? Should they be consistent?